From 75af13ea5be5e72154a8d9cfcb61777ac63dd88c Mon Sep 17 00:00:00 2001
From: Tomas Kopecek <tkopecek@redhat.com>
Date: Mon, 15 Jan 2024 13:48:16 +0100
Subject: [PATCH] Use dnf5-compatible "group install" command

 * yum knows only "yum groupinstall"
 * dnf < 5 knows both
 * dnf-5 only "dnf5 group install"

From RHEL5 version of yum, @group syntax is supported, so we can
replace all groupinstall commands with "install @group".

Related: https://pagure.io/koji/issue/3971
---
 koji/__init__.py                          | 2 +-
 tests/test_cli/test_mock_config.py        | 2 +-
 tests/test_lib/data/mock/internaldev.out  | 2 +-
 tests/test_lib/data/mock/internaldev2.out | 2 +-
 tests/test_lib/data/mock/simple.out       | 2 +-
 5 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/koji/__init__.py b/koji/__init__.py
index b7982abfce84..6fef0f603362 100644
--- a/koji/__init__.py
+++ b/koji/__init__.py
@@ -1592,7 +1592,7 @@ def genMockConfig(name, arch, managed=False, repoid=None, tag_name=None, **opts)
         'target_arch': opts.get('target_arch', arch),
         'chroothome': '/builddir',
         # Use the group data rather than a generated rpm
-        'chroot_setup_cmd': 'groupinstall %s' % opts.get('install_group', 'build'),
+        'chroot_setup_cmd': 'install @%s' % opts.get('install_group', 'build'),
         # don't encourage network access from the chroot
         'rpmbuild_networking': opts.get('use_host_resolv', False),
         'use_host_resolv': opts.get('use_host_resolv', False),
diff --git a/tests/test_cli/test_mock_config.py b/tests/test_cli/test_mock_config.py
index a948aa499f4b..df4e13374e7c 100644
--- a/tests/test_cli/test_mock_config.py
+++ b/tests/test_cli/test_mock_config.py
@@ -28,7 +28,7 @@ class TestMockConfig(utils.CliTestCase):
         }
         self.mock_output = """# Auto-generated by the Koji build system
 
-config_opts['chroot_setup_cmd'] = 'groupinstall build'
+config_opts['chroot_setup_cmd'] = 'install @build'
 config_opts['use_host_resolv'] = False
 config_opts['root'] = 'fedora26-build-repo_1'
 config_opts['yum.conf'] = '[main]\ncachedir=/var/cache/yum\ndebuglevel=1\nlogfile=/var/log/yum.log\nreposdir=/dev/null\nretries=20\nobsoletes=1\ngpgcheck=0\nassumeyes=1\nkeepcache=1\ninstall_weak_deps=0\nstrict=1\n\n# repos\n\n[build]\nname=build\nbaseurl=https://fedora.local/kojifiles/repos/fedora26-build/1/x86_64\n'
diff --git a/tests/test_lib/data/mock/internaldev.out b/tests/test_lib/data/mock/internaldev.out
index 6ef574ec4b54..c99d48dc4fae 100644
--- a/tests/test_lib/data/mock/internaldev.out
+++ b/tests/test_lib/data/mock/internaldev.out
@@ -1,7 +1,7 @@
 # Auto-generated by the Koji build system
 
 config_opts['basedir'] = '/var/lib/mock'
-config_opts['chroot_setup_cmd'] = 'groupinstall build'
+config_opts['chroot_setup_cmd'] = 'install @build'
 config_opts['chroothome'] = '/builddir'
 config_opts['dnf_warning'] = True
 config_opts['internal_dev_setup'] = False
diff --git a/tests/test_lib/data/mock/internaldev2.out b/tests/test_lib/data/mock/internaldev2.out
index c3e88d3748f1..a855153a00da 100644
--- a/tests/test_lib/data/mock/internaldev2.out
+++ b/tests/test_lib/data/mock/internaldev2.out
@@ -1,7 +1,7 @@
 # Auto-generated by the Koji build system
 
 config_opts['basedir'] = '/var/lib/mock'
-config_opts['chroot_setup_cmd'] = 'groupinstall build'
+config_opts['chroot_setup_cmd'] = 'install @build'
 config_opts['chroothome'] = '/builddir'
 config_opts['dnf_warning'] = True
 config_opts['internal_dev_setup'] = True
diff --git a/tests/test_lib/data/mock/simple.out b/tests/test_lib/data/mock/simple.out
index 39866cd224e4..2d621c316f34 100644
--- a/tests/test_lib/data/mock/simple.out
+++ b/tests/test_lib/data/mock/simple.out
@@ -1,7 +1,7 @@
 # Auto-generated by the Koji build system
 
 config_opts['basedir'] = '/var/lib/mock'
-config_opts['chroot_setup_cmd'] = 'groupinstall build'
+config_opts['chroot_setup_cmd'] = 'install @build'
 config_opts['chroothome'] = '/builddir'
 config_opts['dnf_warning'] = True
 config_opts['root'] = 'ROOTNAME'
-- 
2.53.0

